docs/advanced_topics.rst

```diff
@@ -0,0 +1,43 @@
+Advanced topics
+=================
+
+In this section, you can find some advanced functionalities and topics that are just partially mentioned in the API description. For the typical use case, DALI default configuration should perform well out of the box and you don't need to worry about those.
+
+DALI internal data format
+-------------------------
+
+Internally, DALI processes data in NHWC format. To reshape it to NCHW please use ``\*Permute`` operator family. It also means that if mentioned operators are used in the middle of the pipeline then next operations will take the wrong assumption about processed data format and the result is undefined.
+
+Thread affinity
+---------------
+
+The rationale behind this functionality it to allow pining DALI threads to given CPU. It can save the overhead of worker threads jumping from core to core and increase performance for CPU heave workloads.
+It is possible to set CPU thread affinity for DALI using ``DALI_AFFINITY_MASK`` environment variable. ``DALI_AFFINITY_MASK`` should be a comma separated list of CPU ids numbers that will be assigned to corresponding DALI threads. A number of DALI threads is set during pipeline construction by ``num_threads`` argument, ``set_affinity`` enables setting thread affinity for the CPU worker threads. It is worth to mention that due to performance reason nvJPEG decoder operator houses threads on its own and they are always affined.
+When ``DALI_AFFINITY_MASK`` is empty affinity is set based on the result returned by nvmlDeviceGetCpuAffinity.
+If there are more threads than values in ``DALI_AFFINITY_MASK`` value from nvmlDeviceGetCpuAffinity will be used for a thread without value provided by an environment variable,
+I.e.:
+
+.. code-block:: bash
+
+  num_threads=5
+  DALI_AFFINITY_MASK=3,5,6,10
+
+Will set thread 0 to CPU 3, 1 to CPU 5, 2 to CPU 6, 3 to CPU 10 and 4 to CPU which is returned by nvmlDeviceGetCpuAffinity.
+
+Memory consumption
+------------------
+
+DALI architecture makes all GPU operators processing the whole batch at the time and CPU one sample at the time. As DALI is only enlarging its memory, never shrinking, each GPU buffer is as large as the largest possible batch, while CPU buffers are as large as batch size multiplied by the size of the largest sample (even though CPU processes one sample at the time per thread still vector of samples need to reside in the memory). This lazy allocation strategy helps in cases when memory allocation is expensive like pinned CPU memory or GPU memory reducing the number of total memory operations. This is most visible for the operators which output size may differ from sample to sample, from run to run. Operator with the fixed size outputs - like crop is not affecting overall memory consumption growth over time.
+
```

Comment:
I hope I didn't change the technical meaning:

"In DALI architecture all GPU operators process the whole batch at a time while the CPU operator processes one sample at a time. 

With DALI the memory is only growing, never shrinking. Hence each GPU buffer is as large as the largest possible batch, while the CPU buffers are as large as batch size multiplied by the size of the largest sample. Note that even though the CPU processes one sample at a time per thread, the _vector_ of samples need to reside in the memory. 

This lazy allocation strategy reduces the number of total memory operations, and hence helps in cases when the memory allocation is expensive, like the pinned CPU memory or GPU memory. 

This is most visible for the operators whose output size may differ from sample to sample and from run to run. Operator with the fixed size outputs, such as crop, does not influence the overall memory consumption growth over time."

Reply:
Looks perfect